### PR TITLE
Python3 support

### PR DESCRIPTION
**Pull request by dashesy**
~~~This is still work in progress (things like byte/str is not fixed) but at least `list` works.~~~

### CODE REVIEW

**Comment by nirbheek** [comment]
There's conflicts here now; perhaps because I updated the master branch to upstream. Please fix those.

We don't do development on the Cerbero master branch on GitHub, however we should definitely try to merge this into master upstream. Let's do review here (because it's easier) and then merge it upstream: https://cgit.freedesktop.org/gstreamer/cerbero

Can you please comment on https://bugzilla.gnome.org/show_bug.cgi?id=733067 and point to this PR and your intention to work on it?

Thanks!

**Comment by nirbheek** [comment]
I haven't looked in detail, but a couple of comments:

1) How hard is it, in your opinion, to make Cerbero work with both Python 2 and 3? We also use Cerbero on OS X and it ships with only Python 2 by default, so it would be nice to not require Homebrew or Macports for running on OS X.

2) Please always use descriptive commit messages unless you mean to squash commits together. Also, keep commits that do porting to Python 3 and make other changes separate. In the end, we want one commit that just has changes to port to Python 3 and separate commits for other changes.

**Comment by dashesy** [comment]
Thanks

1) It should work in both Python2 and Python3. For the most part I use idioms that are valid in both versions, for something like `iteritems` I use the `six` package but could as well wrap `iteritems` similarly to what `six` does. The wrapping bits are done, now I am looking at `source` to see why some recipes do not get loaded, for that I need to understand the underlying mechanisms better. Hopefully this weekend. 
2)  I will resolve the conflicts, and squash it to one (only concerning Python3 support). I was not even sure about how to contribute.

**Comment by dashesy** [comment]
Commands like `bootstrap`, `build`, `buildone`, ... seem to be working now for those I tested on Linux, ~~will test on Windows tomorrow~~ and Windows. 

Some unrelated bugs:
1. When I try to run tests (with nosetests) they do not pass, even for Python2. For example, it seems DummyConfig does not have `sysroot` which is needed.
2. The documentations are obsolete, there is no gtk+ recipe, but gtk+3, 
3. When specifying config files .cbc is always appended, so linux.cbc will become linux.cbc.cbc
4. configuration is always searched in ~/.cerbero/ even when an absolute path is given

---

~~In 'gettext-tools' CXX is set to 'no' in Makefile (even though environment variable is set) and on Windows, some recipes hav .cc files that need CXX. It is autotools, ... and I have no idea how to properly fix them, but providing CXX manually will let the compile continue.~~

 ~~Update: If I drop to shell and do `./configure ...` with exactly same arguments, CXX is set and make completes. So I am going to see what is it that makes a difference here. The rest of packages compile fine.~~
Fixing environment I think fixed the above problem.

**Comment by dashesy** [comment]
@nirbheek How should I proceed to upstream these changes?

**Comment by nirbheek** [comment]
Sorry for the late reply. I'll have to review these changes here, and after we've done some back-and-forth and I've done some testing on Windows, Linux, and OS X, we can put it on the bugzilla bug, get some more opinions and then I'll merge it upstream.

This is a pretty big change, so we'll have to be careful. I definitely want to get this in though so it will happen. :)

Thanks for your work on this!

**Comment by isometriq** [comment]
Just a note about the instructions and Python versions...

If followed the instructions and I've ended with Python27 not being in the PATH. So if its a manual step, it should be told in the docs.

Moreover the README.MD refers to this command-line
./cerbero-uninstalled
which is incorrect because it uses "python", when it should be "python2" ?

I've found somewhere else this
python2 cerbero-uninstalled

At the moment I can't build by following the instructions of the README :
`IOError: [Errno 2] No such file or directory: '99-dont-add-msys-paths.sh'`

Please let me know where to relocate my issue.
Thanks, looks promising to build faster in Windows!

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ MANIFEST.in
 *~
 *.wsx
 *.wixobj
+.idea
```
Comment (ocrete) [inline .gitignore:18]: Should be separate

**Review by ocrete — COMMENTED** [review]
Unrelated to Python 3 support

**Review by ocrete — COMMENTED** [review]
Has this been tested on whatever Python version comes with macOS ?

**Pull request closed without merging**